docs: document AI-Q 2.2 candidate scope (#322)
NVIDIA-AI-Blueprints/aiq-research-assistant: docs/source/resources/release-notes.md

<!--
SPDX-FileCopyrightText: Copyright (c) 2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
SPDX-License-Identifier: Apache-2.0
-->

# AI-Q 2.2 Release Notes (Unreleased)

These notes describe the release-candidate scope that is merged on the `develop`
branch. AI-Q 2.2 has not been published: there is no final `v2.2.0` tag, container
image, or Helm chart for this candidate yet.

## Research That Routes, Scales, and Finishes Coherently

The deep-research path now separates advisory source routing, structured planning,
concurrent evidence collection, and final synthesis. A planner produces validated
research queries, researcher workers execute independent queries concurrently and
return structured notes, and a dedicated writer synthesizes the final answer. Source
tools can also accept bounded batches of inputs. A caller's `data_sources` selection
constrains tools mapped in the data-source registry; configured unmapped tools, including

> utilities, remain callable. See [Deep Researcher Agent](../architecture/agents/deep-researcher.md)

Reviewer: ```suggestion
utilities, remain callable. Refer to [Deep Researcher Agent](../architecture/agents/deep-researcher.md)
```

and [Automatic Source Routing](../customization/tools-and-sources.md#automatic-source-routing).

The clarifier still gathers missing context and the requested output shape, but it no
longer creates or asks the user to approve a research plan. Planning is owned by the

> deep-research workflow. See [Clarifier Agent](../architecture/agents/clarifier.md).

Reviewer: ```suggestion
deep-research workflow. Refer to [Clarifier Agent](../architecture/agents/clarifier.md).
```

The focused [`config_domain_routing_and_skills.yml`](https://github.com/NVIDIA-AI-Blueprints/aiq/blob/develop/configs/config_domain_routing_and_skills.yml)
profile demonstrates routed research with DuckDuckGo news, Polymarket, paper search,
LlamaIndex, built-in research and synthesis skills, and Modal execution. It is a
specialized profile, not a universal 2.2 default.

## Continue Working From a Completed Report

Conversation follow-up can now use the most recent completed report as durable context:

- A report question is answered against the parent report without launching new research.
- A cosmetic rewrite creates a child report-edit job in the async web/API path; the parent report is unchanged.
- A request for new evidence runs delta research with the parent report available as context.

> The API also exposes an explicit report-edit endpoint. See [Report Follow-up](../integration/rest-api.md#edit-a-report-report-follow-up)

Reviewer: ```suggestion
The API also exposes an explicit report-edit endpoint. Refer to [Report Follow-up](../integration/rest-api.md#edit-a-report-report-follow-up)
```

for the parent/child contract, authorization behavior, and response fields.

## Skills, Sandboxes, and Durable Outputs

DeepAgents code execution now uses a provider-neutral sandbox contract. Modal creates a
fresh sandbox for each job. The experimental OpenShell profile instead attaches jobs to
one pre-provisioned named sandbox; per-job directories reduce ordinary filename

> collisions but are not multi-tenant isolation. See [Deep Research Sandbox](../architecture/agents/sandbox.md)

Reviewer: ```suggestion
collisions but are not multi-tenant isolation. Refer to [Deep Research Sandbox](../architecture/agents/sandbox.md)
```

and the [Skills and Sandbox example](../examples/skills-sandbox/index.md).

> Generated rich files can be harvested into durable artifact records. Metadata remains in

Reviewer: ```suggestion
Generated rich files can be harvested into durable artifact records. The metadata remains in
```

the job database, while bytes use SQL by default or an opt-in S3-compatible store. The
REST API can list metadata and stream content. With `REQUIRE_AUTH=true`, access is scoped

> to the job's owning principal; the default no-auth mode does not enforce job ownership.

Reviewer: ```suggestion
to the job's owning principal. The default no-auth mode does not enforce job ownership.
```

Artifact images referenced by a report render inline in the web UI. Markdown downloads
rewrite those references to artifact content URLs, while PDF export embeds the resolved
images. The `aiq-research` skill can download all durable artifacts or create a portable
bundle containing `report.md` and an `artifacts/` directory with embedded artifact image

> references rewritten to local files. See [Agent Skills for Coding Harnesses](../integration/agent-skills.md#report-follow-up-and-portable-outputs).

Reviewer: ```suggestion
references rewritten to local files. Refer to [Agent Skills for Coding Harnesses](../integration/agent-skills.md#report-follow-up-and-portable-outputs).
```

> See [REST API](../integration/rest-api.md#durable-sandbox-artifacts) and

Reviewer: Please change all "See..." to "Refer to..."

[Production Artifact Storage](../deployment/production.md#artifact-storage).

AI-Q also ships two consumer Agent Skills: `aiq-deploy` for selecting, starting, and
validating a deployment, and `aiq-research` for routed chat and async research against a
running server. The maintainer skill set has expanded with data-source, tool, release-QA,
PR, prompt/model, and CI workflows. See [Agent Skills for Coding Harnesses](../integration/agent-skills.md).

## Enterprise Sources and Policy Controls

> - **Per-user MCP OAuth:** A signed-in user can connect an OAuth-protected MCP source, see its status, and reconnect when the stored token is missing or expired. Jobs resolve that user's tools from a token store shared by the API and worker. See [Per-User MCP OAuth](../customization/mcp-tools.md#per-user-mcp-oauth).

Reviewer: ```suggestion
- **Per-user MCP OAuth:** A signed-in user can connect an OAuth-protected MCP source, observe its status, and reconnect when the stored token is missing or expired. Jobs resolve that user's tools from a token store shared by the API and worker. Refer to [Per-User MCP OAuth](../customization/mcp-tools.md#per-user-mcp-oauth).
```
